File: tests/07_phases_ompss.dg/fortran/success_task_17.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>

SUBROUTINE FOO(X)
    IMPLICIT NONE
    INTEGER, VALUE :: X
    X = 2
END SUBROUTINE FOO

PROGRAM P
    IMPLICIT NONE
    INTERFACE
        !$OMP TASK
        SUBROUTINE FOO(X)
        IMPLICIT NONE
        INTEGER, VALUE :: X
        END SUBROUTINE FOO
    END INTERFACE

    INTEGER :: X
    X = 0

    CALL FOO(X)
    !$OMP TASKWAIT

    if ( X /= 0) STOP 1
END PROGRAM P

File: tests/07_phases_ompss.dg/cxx/success_task_17.cpp
/*
<testinfo>
test_generator=config/mercurium-ompss
</testinfo>
*/

#include <cstdlib>
#include <map>

int main (int argc, char *argv[])
{
	std::map<int, int> m;

    for (int i = 0; i < 100; i++)
    {
        m[i] = i;
    }

	for (std::map<int, int>::iterator it = m.begin();
            it != m.end();
            it++)
	{
		#pragma omp task
		{
            it->second++;
		}
	}

#pragma omp taskwait

    for (int i = 0; i < 100; i++)
    {
        if (m[i] != (i + 1))
            abort();
    }

    return 0;
}
